This patch deliberately tries to be minimal, it would obviously
be possible to implement map_error in a way that preserved more
information, or even eliminate it completely, but that is not
a change I want to make downstream.

Index: tui/Cargo.toml
===================================================================
--- tui.orig/Cargo.toml
+++ tui/Cargo.toml
@@ -105,7 +105,7 @@ version = "1.3"
 version = "0.3"
 
 [dependencies.crossterm]
-version = "0.25"
+version = "0.27"
 optional = true
 
 [dependencies.serde]
Index: tui/src/backend/crossterm.rs
===================================================================
--- tui.orig/src/backend/crossterm.rs
+++ tui/src/backend/crossterm.rs
@@ -122,7 +122,7 @@ where
     }
 }
 
-fn map_error(error: crossterm::Result<()>) -> io::Result<()> {
+fn map_error(error: io::Result<()>) -> io::Result<()> {
     error.map_err(|e| io::Error::new(io::ErrorKind::Other, e.to_string()))
 }
 
